daemons/base.py

```diff
@@ -23,6 +24,39 @@ def rpc(f):
     return f
 
 
+def authenticate(f):
+    def wrapper(daemon, request):
+        auth = request.headers.get("Authorization")
+        user, password = daemon.decode_auth(auth)
+        if not (user == daemon.LOGIN and password == daemon.PASSWORD):
+            return daemon.send_error_response(code=-32600, message="Unauthorized")
+        return f(daemon, request)
+
+    return wrapper
+
+
+@dataclass
+class JsonResponse:
+    id: int
+    code: Optional[int] = None
+    error: Optional[str] = ""
+    result: Optional[Any] = None
+
+    def send(self):
+        if self.result and self.error:
+            raise ValueError(f"result={self.result} and error={self.result} cannot be both set")
+        if self.result is not None:
+            return self.send_ok_response()
+        else:
+            return self.send_error_response()
+
+    def send_error_response(self):
+        return web.json_response({"jsonrpc": "2.0", "error": {self.code: -32600, "message": self.error}, "id": self.id})
```

Comment:
Did you mean that? Code is int
```suggestion
        return web.json_response({"jsonrpc": "2.0", "error": {"code": self.code, "message": self.error}, "id": self.id})
```

Reply:
oh yeah, good catch!